src/controllers/swapAndBridge/swapAndBridge.ts

```diff
@@ -324,6 +331,39 @@ export class SwapAndBridgeController extends EventEmitter implements ISwapAndBri
 
     // eslint-disable-next-line @typescript-eslint/no-floating-promises
     this.#initialLoadPromise = this.#load()
+
+    this.#updateQuoteInterval = createRecurringTimeout(async () => {
+      if (this.formStatus !== SwapAndBridgeFormStatus.ReadyToSubmit) {
+        this.#updateQuoteInterval.stop()
+        return
+      }
+
+      await this.updateQuote({
+        skipPreviousQuoteRemoval: true,
+        skipQuoteUpdateOnSameValues: false,
+        skipStatusUpdate: false
+      })
+    }, UPDATE_SWAP_AND_BRIDGE_QUOTE_INTERVAL)
+
+    this.#updateActiveRoutesInterval = createRecurringTimeout(async () => {
+      if (!this.activeRoutesInProgress.length) {
+        this.#updateActiveRoutesInterval.stop()
+        return
+      }
+
+      await this.checkForNextUserTxForActiveRoutes()
+
+      const minServiceTime = getActiveRoutesLowestServiceTime(this.activeRoutesInProgress)
+      this.#updateActiveRoutesInterval.updateTimeout({ timeout: minServiceTime })
+    }, UPDATE_SWAP_AND_BRIDGE_QUOTE_INTERVAL)
+
+    this.onUpdate(() => {
+      if (this.activeRoutesInProgress.length) {
+        this.#updateActiveRoutesInterval.start()
+      } else {
+        this.#updateActiveRoutesInterval.stop()
+      }
+    }, 'swap-and-bridge')
```

Comment:
Please do this in the setter `activeRoutes`. Make `activeRoutesInProgress` read from the private variable, update the private variable and then check 
```
    if (this.activeRoutesInProgress) {
      this.#updateActiveRoutesInterval.start()
    } else {
      this.#updateActiveRoutesInterval.stop()
    }
```